### ISSUE TITLE
[Question]: My first fork and my first vscode extension. I want to contribute to the project.

### ISSUE THREAD

**Issue by serranomorante** [rxliuli/joplin-utils#61]
# Confirmation

- [x] I checked [FAQ](https://joplin-utils.rxliuli.com/joplin-vscode-plugin/guide/faq.html)

# The tool/kit related to this post

joplin-vscode-plugin

# Describe the question

Sorry if this is not the place to talk about this.

I already checked this page: [Contribute](https://joplin-utils.rxliuli.com/dev/), but the information is rather scarce for a "beginner" in terms of vscode extension development + pnpm monorepos... so I don't know where to start in this particular project.

I know a lot about typescript, react and vscode, though. The joplin data api doesn't seem difficult to grasp and your source code is understandable by just skimming at it.

I need to know:
- How do you start the Extension Development Host window just for the `joplin-vscode-plugin` in the monorepo. Maybe I just need to change the `.vscode/_launch.json` to `.vscode/launch.json`? and also some of the properties inside it?

Any other suggestion would be appreciate.

I intend to expand the TODO functionality from the `joplin-vscode-plugin` with these simple features:
- adding TODOs from the vscode explorer
- adding a due date for the TODO when creating it
- (maybe) presenting a list showing only the pending TODOs

I don't know why this feature isn't in the roadmap, if there's a reason for which I should avoid continuing with this fork just let me know. Also let me know if you need help with something else.

Btw, I love this project. Thank you so much!

**Comment by rxliuli** [comment]
> How do you start the Extension Development Host window just for the joplin-vscode-plugin in the monorepo. Maybe I just need to change the .vscode/_launch.json to .vscode/launch.json? and also some of the properties inside it?

1. fork repo
2. git clone https://github.com/<YOUR GITHUB NAME>/joplin-utils.git && cd joplin-utils/
3. pnpm i && pnpm setup-all
4. code apps/joplin-vscode-plugin
5. run joplin dev mode, ref: <https://joplinapp.org/api/references/development_mode/>
  ![image](https://user-images.githubusercontent.com/24560368/198864483-b30f050f-e990-4a49-868a-2954eea75443.png)
6. enable clipper service
  ![image](https://user-images.githubusercontent.com/24560368/198864546-473dbc9b-6f09-4cf5-8585-87da13b8b039.png)
7. cp .vscode/_launch.json .vscode/launch.json
8. modify .vscode/launch.json, set _env.JOPLIN_TOKEN_, example: 
  ![image](https://user-images.githubusercontent.com/24560368/198864435-ac47e951-79ad-40c5-b848-9e5dbafad478.png)
9. F5 run debug

> I don't know why this feature isn't in the roadmap, if there's a reason for which I should avoid continuing with this fork just let me know. Also let me know if you need help with something else.

Simply put, I don't use the joplin todo feature and I don't think it's a really usable todo app.

**Comment by serranomorante** [comment]
I'm getting this error inside the Extension Development Host window after activating the extension using `View: Show Joplin`:

![View Joplin](https://i.postimg.cc/NFrP9hc5/vmware-20221030-020227-997.png)

I did all the steps that you mentioned.

I'm using Ubuntu. I installed vscode using snap.

This is the stdout of running `pnpm i && pnpm setup-all`

```python
Scope: all 11 workspace projects
Lockfile is up to date, resolution step is skipped
Already up to date
. postinstall$ npx simple-git-hooks
│ [INFO] Successfully set the pre-commit with command: pnpm lint-staged
│ [INFO] Successfully set the commit-msg with command: pnpm commitlint --edit $1
│ [INFO] Successfully set all git hooks
└─ Done in 1s
Done in 5.9s

> joplin-utils@ setup-all /home/<username>/contributions/joplin-utils
> pnpm -r run setup

Scope: 10 of 11 workspace projects
libs/joplin-api setup$ pnpm build
│ > joplin-api@0.5.0 build /home/<username>/contributions/joplin-utils/libs/joplin-api
│ > rimraf dist && tsup src/index.ts src/node-polyfill.ts --format esm,cjs --dts
│ CLI Building entry: src/index.ts
[1 lines collapsed]
│ CLI tsup v6.2.3
│2CLI Target: node14
│ CLI Building entry: src/index.ts
│ CLI Using tsconfig: tsconfig.json
│3CLI tsup v6.2.3
│ CLI Using tsconfig: tsconfig.json
│ CLI tsup v6.2.3
│4CLI Target: node1413.30 KB
│ CLI tsup v6.2.3
│ CLI Target: node14ss in 163ms
│5ESM Build start
│ CLI Target: node14ss in 163ms
│ ESM Build start
│ CJS Build startccess in 163ms
│ CJS dist/index.cjs 15.20 KB
│ CJS ⚡️ Build success in 163ms
│ ESM dist/index.js 13.30 KB
│ ESM ⚡️ Build success in 197mss
│ DTS Build start
│ DTS ⚡️ Build success in 4476ms
└─ Done in 6.9sx.d.ts 17.29 KB
apps/joplin-blog setup$ pnpm run gen-i18n && pnpm run build
│ > joplin-blog@0.9.4 gen-i18n /home/<username>/contributions/joplin-utils/apps/joplin-blog
│ > i18next-dts-gen --dirs src/i18n
│ Ends generating type definition [79ms]: src/i18n/index.d.ts
│ > joplin-blog@0.9.4 build /home/<username>/contributions/joplin-utils/apps/joplin-blog
│ > tsup src/index.ts --format esm,cjs --dts --shims && tsup src/bin.ts --format esm --target esnext
│ CLI Building entry: src/index.ts
[1 lines collapsed]
│ CLI tsup v6.2.3
│ CLI Target: node14
│ ESM Build start
└─ Running...
a2ps/joplin-vscode-plugin setup$ i18next-dts-gen -i src/i18n/
│ Ends generating type definition [79ms]: src/i18n/index.d.ts
│ > joplin-blog@0.9.4 build /home/<username>/contributions/joplin-utils/apps/joplin-blog
│3> tsup src/index.ts --format esm,cjs --dts --shims && tsup src/bin.ts --format esm --target esnext
│ > joplin-blog@0.9.4 build /home/<username>/contributions/joplin-utils/apps/joplin-blog
│ > tsup src/index.ts --format esm,cjs --dts --shims && tsup src/bin.ts --format esm --target esnext
│4CLI Building entry: src/index.ts
│ > tsup src/index.ts --format esm,cjs --dts --shims && tsup src/bin.ts --format esm --target esnext
│ CLI Building entry: src/index.ts
│5CLI Using tsconfig: tsconfig.json
│ CLI Building entry: src/index.ts
│ CLI Using tsconfig: tsconfig.json
│6CLI tsup v6.2.3
│ CLI Using tsconfig: tsconfig.json
│ CLI tsup v6.2.3
│7CLI Target: node14ss in 166ms
│ CLI tsup v6.2.3
│ CLI Target: node14ss in 166ms
│8ESM Build start
│ CLI Target: node14ss in 166ms
│ ESM Build start
│9CJS Build startccess in 166ms
│ ESM Build start
│ CJS Build startccess in 166ms
[10 lines collapsed]16.76 KBB
│ CJS Build startccess in 166mss
│ ESM dist/index.js 16.76 KBB
│ 1SM ⚡️ Build success in 166mss
│ ESM dist/index.js 16.76 KBB
│ ESM ⚡️ Build success in 166mss
│ 2JS dist/index.cjs 19.72 KB
│ ESM ⚡️ Build success in 166mss
│ CJS dist/index.cjs 19.72 KB
│ 3JS ⚡️ Build success in 130mss
│ CJS dist/index.cjs 19.72 KB
│ CJS ⚡️ Build success in 130mss
│ 4TS Build start
│ CJS ⚡️ Build success in 130mss
│ DTS Build start
│ 5TS ⚡️ Build success in 9043ms
│ DTS Build start
│ DTS ⚡️ Build success in 9043ms
│ DTS dist/index.d.ts 5.70 KB
│ CLI Building entry: src/bin.ts
│ CLI Using tsconfig: tsconfig.json
│ CLI tsup v6.2.3
└─ Done in 14.4snext.95 KB
│ ESM Build start
│ ESM dist/bin.js 31.95 KB
│ ESM ⚡️ Build success in 136ms
```

**Comment by rxliuli** [comment]
Sorry, looks like I missed a step, need to run `pnpm dev` before `F5 run debug`

**Comment by serranomorante** [comment]
> Sorry, looks like I missed a step, need to run `pnpm dev` before `F5 run debug`

Thank you!
